### PR TITLE
Set Other Environment Types

### PR DESCRIPTION
**Pull request by bnippercg**
## what
- Support the ability for the codebuild module in terraform to set other environment types to support all types of containers as listed in the AWS codebuild spec.
- Support the ability to customize VPC
- Include IAM for EC2

## why
- To enable the ability to have AWS pipelines implemented using all types of containers as supported by AWS codebuild. 
- Support customize of VPC
- Included needed IAM for EC2

## references
- If accepted this will support closing https://github.com/cloudposse/terraform-aws-codebuild/issues/56
- Will also include features for VPC support

### CODE REVIEW

**Comment by bnippercg** [comment]
Not sure what to do about getting reviewers assigned.

**Comment by bnippercg** [comment]
Ops... sorry didn't mean to close this.

**Comment by osterman** [comment]
(we're not able to rebuild the readme automatically due to permissions on your repo)

**Comment by osterman** [comment]
Also, please fix merge conflicts

**Comment by bnippercg** [comment]
I believe all merge conflicts have been addressed including updating the README. If there are another changes needed please let me know.

Thank you for taking a look at this @osterman

**Comment by osterman** [comment]
@bnippercg - sorry - we need your help to rebase again! We merged another PR before this one.

**Comment by Gowiem** [comment]
@bnippercg Sorry, but looks like some merge conflicts + merge conflicts got checked in. Can you address and we'll get this shipped?

File: docs/targets.md [hunk 1]
```diff
@@ -5,6 +5,6 @@ Available targets:
   help                                Help screen
   help/all                            Display help for all targets
   help/short                          This help short screen
-  lint                                Lint terraform code
+  lint:                              Lint terraform code
```
Comment (Gowiem) [inline docs/targets.md:8]: @bnippercg Can you remove this change while fixing your merge conflicts too?

File: docs/terraform.md [hunk 2]
```diff
@@ -35,11 +35,21 @@
 
 | Name | Description |
 |------|-------------|
+<<<<<<< Updated upstream
```
Comment (Gowiem) [inline docs/terraform.md:38]: @bnippercg This is the merge conflict that got checked in. Please address.

File: variables.tf [hunk 3]
```diff
@@ -172,3 +172,84 @@ variable "report_build_status" {
...
+  description = "A version of the build input to be built for this project. If not specified, the latest version is used."
+}
+
+variable "fetch_git_submodules" {
+  type        = bool
+  default     = false
+  description = "If set to true, fetches Git submodules for the AWS CodeBuild build project."
+}
+
+variable "vpc_config" {
```
Comment (Gowiem) [inline variables.tf:239]: @bnippercg Doesn't look like this variable is used. Is it needed?

File: variables.tf [hunk 4]
```diff
@@ -172,3 +172,84 @@ variable "report_build_status" {
...
+  description = "The Bitbucket username when the authType is BASIC_AUTH. This parameter is not valid for other types of source providers or connections."
+}
+
+variable "source_version" {
+  type        = string
+  default     = ""
+  description = "A version of the build input to be built for this project. If not specified, the latest version is used."
+}
+
+variable "fetch_git_submodules" {
```
Comment (Gowiem) [inline variables.tf:233]: This and a few other variables aren't used... Any reason they around?

File: docs/targets.md (hunk 1 above)
Comment (bnippercg) [reply docs/targets.md:8]: Sure thing. I'll update that with the next push.

File: docs/terraform.md (hunk 2 above)
Comment (bnippercg) [reply docs/terraform.md:38]: Not sure how I missed that, I'll be sure to correct that before the next update.

File: variables.tf (hunk 3 above)
Comment (bnippercg) [reply variables.tf:239]: I think I pasted in something by mistake as I'm not sure where this variable came from. It'll be gone with the next push.

File: variables.tf (hunk 4 above)
Comment (bnippercg) [reply variables.tf:233]: This will be gone with next push. I think it was a cut and past mistake.

Comment (bnippercg) [reply variables.tf]: Actually I did find the reference to this... looks like I missed a merge... now that I got that sorted I'm finding where I introduced this variable.

It'll be there after my next push.

File: variables.tf (hunk 3 above)
Comment (bnippercg) [reply variables.tf]: Actually I did find the reference to this... looks like I missed a merge... now that I got that sorted I'm finding where I introduced this variable.

It'll be there after my next push.

File: variables.tf (hunk 4 above)
Comment (bnippercg) [reply variables.tf]: Ugh... hate when I make a merge mistake. Anyway... I believe this has been sorted out now.

File: variables.tf (hunk 3 above)
Comment (bnippercg) [reply variables.tf]: Ok, I think this is sorted out now.

**Comment by Gowiem** [comment]
@bnippercg Looking at your diff, it looks like the change to targets.md is still in and your vpc updates are gone. The only real change I see is the addition of one policy for `ec2:*`. Is that what you intended?

**Comment by bnippercg** [comment]
> @bnippercg Looking at your diff, it looks like the change to targets.md is still in and your vpc updates are gone. The only real change I see is the addition of one policy for `ec2:*`. Is that what you intended?

HA... I was so mixed up after the series of corrections I had to make from the merge I didn't realize that it appears someone else already implemented the changes that I was looking to make.

Apologizes for not paying closer attention to what others had done.

Thank you for attempting to help me work through the review process though. I hope I get a chance to contribute in the future.

**Comment by Gowiem** [comment]
@bnippercg All good, definitely contribute again in the future and we'll get your work merged. Thanks!

**Pull request closed without merging**